src/types.ts

```diff
@@ -0,0 +1,52 @@
+interface ValidateTextOptionsBase {
+	medium?: 'all' | 'braille' | 'embossed' | 'handheld' | 'print' | 'projection' | 'screen' | 'speech' | 'tty' | 'tv';
+}
+
+type ValidateTextOptionsWithBaseWarning = ValidateTextOptionsBase & {
+	warningLevel?: 0;
+};
+
+type ValidateTextOptionsWithElevatedWarning = ValidateTextOptionsBase & {
```

Comment:
Similarly:

```suggestion
type ValidateTextOptionsWithWarnings = ValidateTextOptionsBase & {
```

...this way, you'd have the phrases "with warnings" and "without warnings." This would match the naming later on of `ValidateTextResultWithWarnings`.